### PR TITLE
nnbd

### PR DESCRIPTION
**Pull request by frankpepermans**
non-nullability branch

*do not merge*

### CODE REVIEW

**Comment by hoc081098** [comment]
Hey @frankpepermans I just:
- comment tests uses mockito and make all the tests pass
- remove unnecessary nullable type, null-assertion

**Comment by hoc081098** [comment]
I used `_Optional<T>` in `BehaviorSubject<T>` to keep latest value even `T` is nullable-type). I think we can refactor `ValueStream` like this:
```dart
class Optional<T> { // or Wrapper<T> :)
    final T value;
}

abstract class ValueStream<T> extends Stream<T> {
    Optional<T>? value; // null when has no value, otherwise we can extract `value` (even value is `null`)
    T? error; // null when has no error
}

ValueStream<int?> s1 = BehaviorSubject<int?>.seeded(null);
if (s1.value == null) print('Has no value');
else print('Latest value is ${s1.value.value}'); // prints: Latest value is null

ValueStream<int?> s2 = BehaviorSubject<int?>();
if (s2.value == null) print('Has no value'); // prints: Has no value
else print('Latest value is ${s2.value.value}');
```

File: pubspec.yaml [hunk 1]
```diff
@@ -10,12 +10,11 @@ description: >
 homepage: https://github.com/ReactiveX/rxdart
 
 environment:
-  sdk: '>=2.6.0 <3.0.0'
+  sdk: '>=2.10.0-56.0.dev <2.11.0'
```
Comment (felangel) [inline pubspec.yaml:13]:
```suggestion
  sdk: ">=2.12.0-0 <3.0.0"
```

**Comment by felangel** [comment]
Are there any updates regarding how soon we can expect a `nullsafety` release on pub.dev?

File: pubspec.yaml (hunk 1 above)
Comment (felangel) [reply pubspec.yaml:13]: opened a PR https://github.com/ReactiveX/rxdart/pull/529

**Comment by frankpepermans** [comment]
@felangel heya! well, still working on it, as soon as all tests are passing again, we can do a -nullsafety.0 release
@hoc081098 I've just merged master into this one, and updated null safety where I could, but we now have about 8 tests failing, still need to check to see why

**Comment by felangel** [comment]
@frankpepermans sounds good! I’ll try to take a look later today and see if I can help fix the failing tests 👍

**Comment by frankpepermans** [comment]
It's probably mistakes I made from merging, there were conflicts all over the place.
I'm also taking another look tomorrow, getting late over here :)

File: pubspec.yaml [hunk 2]
```diff
@@ -10,12 +10,12 @@ description: >
 homepage: https://github.com/ReactiveX/rxdart
 
 environment:
-  sdk: '>=2.6.0 <3.0.0'
+  sdk: '>=2.12.0-51.0.dev <3.0.0'
```
Comment (felangel) [inline pubspec.yaml:13]:
```suggestion
  sdk: '>=2.12.0-0 <3.0.0'
```

**Comment by hoc081098** [comment]
Hey please see https://github.com/ReactiveX/rxdart/tree/hoc081098/nnbd. All tests passed

**Comment by frankpepermans** [comment]
Nice job!

Let's try to prepare a nullsafety release today then, I think a lot of people are waiting on rxdart

**Comment by hoc081098** [comment]
Published first nnbd version: https://pub.dev/packages/rxdart/versions/0.26.0-nullsafety.0

**Comment by hoc081098** [comment]
Published https://pub.dev/packages/rxdart/versions/0.26.0-nullsafety.1

**Pull request merged**